src/python/grpcio_tests/tests/unit/_6891_test.py

```diff
@@ -0,0 +1,204 @@
+        return _SMALL_MESSAGE
+    elif message_size is _MessageSize.LARGE:
+        return _LARGE_MESSAGE
+
+
+def _messages_to_send(count, size):
+    for _ in range(count.value[0]):
+        yield _message(size)
+
+
+def _draw_requests(case, request_iterator):
+    for _ in range(
+            min(case.request_count.value[0], case.request_reading.value[0])):
+        next(request_iterator)
+
+
+def _draw_responses(case, response_iterator):
+    for _ in range(case.response_count.value[0]):
+        next(response_iterator)
+
+
+class _MethodHandler(grpc.RpcMethodHandler):
+
+    def __init__(self, case):
+        self.request_streaming = True
+        self.response_streaming = True
+        self.request_deserializer = None
+        self.response_serializer = None
+        self.unary_unary = None
+        self.unary_stream = None
+        self.stream_unary = None
+        self._case = case
+
+    def stream_stream(self, request_iterator, servicer_context):
+        _draw_requests(self._case, request_iterator)
+
+        for response in _messages_to_send(self._case.response_count,
+                                          self._case.response_size):
+            yield response
+
+
+class _GenericHandler(grpc.GenericRpcHandler):
+
+    def __init__(self, case):
+        self._case = case
+
+    def service(self, handler_call_details):
+        return _MethodHandler(self._case)
+
+
+class _Issue6891Test(unittest.TestCase):
+
+    def setUp(self):
+        self._server = test_common.test_server()
+        port = self._server.add_insecure_port('[::]:0')
+        self._server.add_generic_rpc_handlers((_GenericHandler(self.case),))
+        self._server.start()
+
+        self._channel = grpc.insecure_channel('localhost:%d' % port)
+        self._multi_callable = self._channel.stream_stream(_RPC_METHOD)
+
+    def tearDown(self):
+        self._server.stop(None)
+
+    def test_early_ok(self):
+        requests = _messages_to_send(self.case.request_count,
+                                     self.case.request_size)
+
+        response_iterator_call = self._multi_callable(requests)
+
+        _draw_responses(self.case, response_iterator_call)
+
+        self.assertIs(grpc.StatusCode.OK, response_iterator_call.code())
+
+
+def _cases():
+    for request_count in _MessageCount:
+        for request_size in _MessageSize:
+            for request_reading in _ReadRequests:
+                for response_count in _MessageCount:
+                    for response_size in _MessageSize:
+                        yield _Case(request_count, request_size,
+                                    request_reading, response_count,
+                                    response_size)
+
+
+def _test_case_classes():
+    for case in _cases():
+        yield type(case.create_test_case_name(), (_Issue6891Test,), {
+            'case': case,
+            '__module__': _Issue6891Test.__module__,
+        })
+
+
+def load_tests(loader, tests, pattern):
```

Comment:
I'm not sure how the loader functions work internally.  Can you confirm it works?  

FWIW, I'd prefer to just dump all this in a single unit test, and maybe generate some failure string on the "assertIs" call so we know which part of the test matrix failed.

Reply:
Yes, this test works in Google's internal continuous integration.

Hmm... true, and it might even run measurably faster if the server and channel were shared among the individual cases. But this is good enough (and when it fails the particular cell _should_ be readily apparent) and I think it's worth getting the ideas page edited today. Sorry to brush you off with an "it can happen later" knowing that later will never come, but... it can happen later, when all the other TODOs for the test corpus get addressed. :-\